### PR TITLE
Update extension name in plugin.xml

### LINKED ISSUE #1099: Update Cody extension name for SEO
See https://docs.google.com/document/d/1Dd6IegY0l7g2x2xBPFKFuEtKw6U0F6uYGAJKTK6AwUQ/edit#heading=h.ubvxmq40v7u1

### PR DESCRIPTION
**Pull request by iskyOS**
## Test plan

Metadata name change (for the marketplace) only. No specific tests.

### CODE REVIEW

**Review by taylorsperry — APPROVED** [review]
Thank you, Alex!

**Review by vdavid — APPROVED** [review]
A bit scary to remove "Sourcegraph" from the name but I'm sure you've thought about it.

**Comment by pkukielka** [comment]
> A bit scary to remove "Sourcegraph" from the name but I'm sure you've thought about it.

Vendor will be still showing up, as it is defined in a separate filed:

image

image

**Review by pkukielka — APPROVED** [review]
LGTM

File: src/main/resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <idea-plugin require-restart="true">
     <id>com.sourcegraph.jetbrains</id>
-    <name>Sourcegraph Cody + Code Search</name>
+    <name>Cody: AI Coding Assistant with Autocomplete & Chat</name>
```
Comment (pkukielka) [inline src/main/resources/META-INF/plugin.xml]:
```
Execution failed for task ':patchPluginXml'.
> org.jdom2.input.JDOMParseException: Error on line 3: The entity name must immediately follow the '&' in the entity reference.
```

Unfortunately `&` will not work in xml file, it needs to be replaced with `&amp;`.
I will push fix if I can.

**Pull request merged**